test/unit/user_test.rb

```diff
@@ -119,4 +119,27 @@ def test_create_user_successful_service_not_require_registered_users
 
     assert_equal metrics, user.load_metric_names
   end
+
+  test '.delete_all removes all the Users of a Service' do
+    Service.save!(id: 7003, provider_key: 'test_provkey', default_service: false)
+    Service.save!(id: 7004, provider_key: 'test_provkey', default_service: false)
+    num_users = 600
```

Comment:
600 looks like an arbitrary number. I guess you decided this to be larger than the number of users to be deleted on each batch? If so, I think it would be better for readability to relate this variable with that constant, something like `num_users = BATCH_SIZE + 1`.
What do you think?

Reply:
:+1: . It was indeed to be larger than the number of users to be deleted on each batch